addons/knobs/src/index.js

```diff
@@ -1,66 +1,22 @@
 import { window } from 'global';
+import deprecate from 'util-deprecate';
 import addons from '@storybook/addons';
-import KnobManager from './KnobManager';
+
 import { vueHandler } from './vue';
 import { reactHandler } from './react';
 
-const manager = new KnobManager();
-
-export function knob(name, options) {
-  return manager.knob(name, options);
-}
-
-export function text(name, value) {
-  return manager.knob(name, { type: 'text', value });
-}
-
-export function boolean(name, value) {
-  return manager.knob(name, { type: 'boolean', value });
-}
-
-export function number(name, value, options = {}) {
-  const defaults = {
-    range: false,
-    min: 0,
-    max: 10,
-    step: 1,
-  };
-
-  const mergedOptions = { ...defaults, ...options };
-
-  const finalOptions = {
-    ...mergedOptions,
-    type: 'number',
-    value,
-  };
+import { knob, text, boolean, number, color, object, array, date, manager } from './base';
 
-  return manager.knob(name, finalOptions);
-}
-
-export function color(name, value) {
-  return manager.knob(name, { type: 'color', value });
-}
-
-export function object(name, value) {
-  return manager.knob(name, { type: 'object', value });
-}
+export { knob, text, boolean, number, color, object, array, date };
 
-export function select(name, options, value) {
-  return manager.knob(name, { type: 'select', options, value });
-}
-
-export function array(name, value, separator = ',') {
-  return manager.knob(name, { type: 'array', value, separator });
-}
-
-export function date(name, value = new Date()) {
-  const proxyValue = value ? value.getTime() : null;
-  return manager.knob(name, { type: 'date', value: proxyValue });
-}
+deprecate(
+  () => {},
+  'Using @storybook/knobs directly is discouraged, please use @storybook/knobs/dist/{{framework}}'
```

Comment:
Addons currently adopt the following es5 pattern for registration `/register.js`:

```js
require('./dist/register')
```

Can we do something analogous for frameworks?